web/src/components/organisms/Settings/Members/hooks.ts

```diff
@@ -100,30 +109,28 @@ export default ({ workspaceId }: Props) => {
       .sort((user1, user2) => stringSortCallback(user1.userId, user2.userId));
   }, [currentWorkspace, searchTerm]);
 
-  const [addUserToWorkspaceMutation] = useAddUserToWorkspaceMutation();
-
-  const handleMemberAddToWorkspace = useCallback(
-    async (userIds: string[]) => {
-      const results = await Promise.all(
-        userIds.map(async userId => {
-          if (!workspaceId) return;
-          const result = await addUserToWorkspaceMutation({
-            variables: { userId, workspaceId, role: Role.Reader },
-            refetchQueries: ["GetWorkspaces"],
-          });
-          const workspace = result.data?.addUserToWorkspace?.workspace;
-          if (result.errors || !workspace) {
-            Notification.error({ message: t("Failed to add one or more members.") });
-            return;
-          }
-          setWorkspace(workspace);
-        }),
-      );
-      if (results) {
-        Notification.success({ message: t("Successfully added member(s) to the workspace!") });
-      }
-    },
-    [workspaceId, addUserToWorkspaceMutation, setWorkspace, t],
+  const [addUsersToWorkspaceMutation] = useAddUsersToWorkspaceMutation();
+
+  const handleUsersAddToWorkspace = useCallback(
+    (users: MemberInput[]) =>
+      (async () => {
+        if (!workspaceId) return;
+        const result = await addUsersToWorkspaceMutation({
+          variables: { workspaceId, users: users as GQLMemberInput[] },
+          refetchQueries: ["GetWorkspaces"],
+        });
+        const workspace = result.data?.addUsersToWorkspace?.workspace;
+        if (result.errors || !workspace) {
+          Notification.error({ message: t("Failed to add one or more members.") });
+          return;
+        }
+        setWorkspace(workspace);
```

Comment:
We handle workspace changing (and project changing) basically exclusively in the Header now, so please remove this.
Anyways, it makes no sense to change the workspace on user add.

Reply:
Done!
I did not add this, it was in the original code, I removed it anyways.

Reply:
Sorry I know it is a little out of scope for this PR, but can you also remove all use of `setWorkspace` from this file and see if everything still works ok? 
Also, can you remove the passed workspaceId prop (and its useParam in the index.tsx) and use the `currentWorkspace.id` instead?

Reply:
After doing this refetching does not work anymore. somehow using setWorkspace is triggering refetching the table members. can we handle this in a separate PR?

I used `currentWorkspace.id` instead of `workspaceId`

Reply:
Yeah that isn't a good way to handle the refetch... Can you make sure you handle this in a different PR? You can close this convo after you read this.

Reply:
Sure, I will handle this in a different PR, I don't think that it will be soon though.